src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -145,9 +145,10 @@ private async Task<Frame> ReadFrameAsync(CancellationToken cancellationToken)
                 throw new Exception("Connection stream closed while attempting to read frame body.");
             }
 
-            if (_ignoreSettingsAck && header.Type == FrameType.Settings && header.Flags == FrameFlags.Ack)
+            if (_ignoredSettingsAckPromise != null && header.Type == FrameType.Settings && header.Flags == FrameFlags.Ack)
             {
-                _ignoreSettingsAck = false;
+                _ignoredSettingsAckPromise?.TrySetResult(false);
```

Comment:
Why `?.`? You've already checked it for null... is it expected to be a multi-thread race?

Reply:
It's not really meant to be thread safe, so can remove.